vm-setup/roles/v1aX_integration_test/filter_plugins/status.py

```diff
@@ -0,0 +1,70 @@
+""" filtering k8s_info resources """
+
+from ansible.errors import AnsibleFilterError
+from ansible.utils.display import Display
+
+display = Display()
+
+
+def msg(key, resources):
+    return "could not find %r key in %r" % (key, resources)
+
+
+def filter_phase(resources, phase, skip_missing=True):
```

Comment:
It isn't clear to me what all the different `phase` values I would use here are. Could we make this simpler? I'd rather all these filters take only one argument and then do what they say on the filter name, instead than doing logic based on strings from Jinja.

Reply:
Agree ! that was the idea in the beginning, however going through the different tasks different `phase` was needed such as `running`, `provisioning` and `deleting` creating a filter for each end up duplicating the same code which is a bit difficult to maintain moreover we will get confusing with other filter `phase_ready`, `provisioning_ready` in the provisioning filter. and when` ready = True / False`  So I still believe sticking to the json behavior and having small number of parameterized filters will be more efficient in the future to use, maintain, and easy to understand

Reply:
In my opinion, duplication isn't bad when it makes the code easy to use. However, I can see that you have thought about this. It is not a blocker for me. I think it would be helpful in the docstring to give some example of usage and more details about the inputs that we expect to use.